File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack = new Stack<>();

        int[] output = new int[temperatures.length];
        
        for(int i = 0; i < temperatures.length; i++) {
            while(!stack.isEmpty() && (temperatures[i] > temperatures[stack.peek()])) {
                int prevIndex = stack.pop();
                int numDaysWaited = i - prevIndex;
                output[prevIndex] = numDaysWaited;
            }
            stack.push(i);
        }

        return output;
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        
        myStack = []
        myHashMap = {'}' : '{', ']' : '[', ')' : '('}

        for char in s:
            if char in myHashMap:
                if myStack and myStack[-1] == myHashMap[char]:
                    myStack.pop()
                else:
                    return False
            else:
                myStack.append(char)
        
        if not myStack:
            return True
        else:
            return False
